legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/pureToSQLQuery/pureToSQLQuery.pure

```diff
@@ -2721,8 +2725,9 @@ function meta::relational::functions::pureToSqlQuery::processDrop(f:FunctionExpr
    let mainQuery = processValueSpecification($f.parametersValues->at(0), $currentPropertyMapping, $operation, $vars, $state, $joinType, $nodeId, $aggFromMap, $context, $extensions)->toOne()->cast(@SelectWithCursor);
    let mainSelect = $mainQuery.select;
 
-   let fromValue = processValueSpecification($f.parametersValues->at(1), $currentPropertyMapping, $operation, $vars, $state, JoinType.LEFT_OUTER, $nodeId, $aggFromMap, $context, $extensions)->toOne()->cast(@SelectWithCursor).select.columns->at(0)->cast(@Literal);//.value->cast(@Integer);
-
+   let param1 = processValueSpecification($f.parametersValues->at(1), $currentPropertyMapping, $operation, $vars, $state, JoinType.LEFT_OUTER, $nodeId, $aggFromMap, $context, $extensions)->toOne()->cast(@SelectWithCursor).select.columns->at(0);//.value->cast(@Integer);
+   assert($param1->instanceOf(Literal),'Only literals are supported inside the drop function, you are passing a ' + $param1->type()->toString());
+   let fromValue = $param1->cast(@Literal);
```

Comment:
No tests ?

Reply:
I am unaware of any way we could write failure tests in pure i.e. asserting on exception

Reply:
Since this is a small non-functional change I feel existing tests should be enough to confirm it as a non-breaking change, though if we still need tests I will write those in engine. Thoughts?

Creating assertFail sounds reasonable to me.